packages/api/src/controllers/access-control.ts

```diff
@@ -40,83 +44,181 @@ app.post(
+          console.log(`
+            access-control: gate: content contentId=${content.id} with playbackId=${playbackId} is gated but multiple (${signingKeyOutput.length}) public keys found for key=${req.body.pub}, disallowing playback, colliding keys=${collisionKeys}
+          `);
+          throw new BadRequestError(
+            "Multiple signing keys found for the same public key."
+          );
+        }
+
+        const signingKey = signingKeyOutput[0];
+
+        if (signingKey.userId !== content.userId) {
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the content and the public key pub=${req.body.pub} do not share the same owner, signingKeyUserId=${signingKey.userId}, contentUserId=${content.userId}
+          `);
+          throw new NotFoundError("Content not found");
+        }
+
+        if (signingKey.disabled || signingKey.deleted) {
+          const signingKeyLog = JSON.stringify(JSON.stringify(signingKey));
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the public key pub=${signingKey.id} is disabled or deleted, signingKey=${signingKeyLog}
+          `);
+          throw new ForbiddenError("The public key is disabled or deleted");
+        }
+
+        tracking.recordSigningKeyValidation(signingKey.id);
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+        res.status(204);
+        return res.end();
+      case "webhook":
+        if (!req.body.accessKey || req.body.type !== "accessKey") {
+          throw new ForbiddenError(
+            "Content is gated and requires an access key"
+          );
+        }
+        const webhook = await db.webhook.get(content.playbackPolicy.webhookId);
+        if (!webhook) {
+          console.log(`
+            access-control: gate: content with playbackId=${playbackId} is gated but corresponding webhook not found for webhookId=${content.playbackPolicy.webhookId}, disallowing playback
+          `);
+          throw new InternalServerError(
+            "Content is gated and corresponding webhook not found"
+          );
+        }
+        const statusCode = await fireGateWebhook(
+          webhook,
+          content.playbackPolicy,
+          req.body.accessKey
         );
-      }
-
-      if (signingKeyOutput.length > 1) {
-        let collisionKeys = JSON.stringify(signingKeyOutput);
-        console.log(`
-          access-control: gate: content contentId ${content.id} with playbackId=${playbackId} is gated but multiple (${signingKeyOutput.length}) public keys found for key ${req.body.pub}, disallowing playback, colliding keys=${collisionKeys}
-        `);
+        if (statusCode >= 200 && statusCode < 300) {
+          res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+          res.status(204);
+          return res.end();
```

Comment:
This logic could also be shared with the `public` playback policy by going in the end of the function instead. So it's like each `case` here should validate the access conditions and throw an error if it fails, otherwise let it continue the handler and allow playback. WDYT?